apps/rush-lib/src/logic/deploy/DeployManager.ts

```diff
@@ -133,85 +136,131 @@ export class DeployManager {
         } catch (resolveErr) {
-          if (resolveErr.code === 'MODULE_NOT_FOUND' && optionalDependencyNames.has(dependencyPackageName)) {
-            // Ignore missing optional dependency
-            continue;
+          if (resolveErr.code === 'MODULE_NOT_FOUND') {
+            // The workaround link isn't guaranteed to exist, so ignore if it's missing
+            // NOTE: If you encounter this warning a lot, please report it to the Rush maintainers.
+            console.log('Ignoring missing PNPM workaround link for ' + packageJsonFolderPath);
           }
-          throw resolveErr;
         }
       }
+    }
+  }
 
-      if (
-        this._rushConfiguration.packageManager === 'pnpm' &&
-        !deployState.scenarioConfiguration.json.omitPnpmWorkaroundLinks
-      ) {
-        // Replicate the PNPM workaround links.
-
-        // Only apply this logic for packages that were actually installed under the common/temp folder.
-        if (Path.isUnder(packageJsonFolderPath, this._rushConfiguration.commonTempFolder)) {
-          try {
-            // The PNPM workaround links are created in this folder.  We will resolve the current package
-            // from that location and collect any additional links encountered along the way.
-            const pnpmDotFolderPath: string = path.join(
-              this._rushConfiguration.commonTempFolder,
-              'node_modules',
-              '.pnpm'
-            );
-
-            // TODO: Investigate how package aliases are handled by PNPM in this case.  For example:
-            //
-            // "dependencies": {
-            //   "alias-name": "npm:real-name@^1.2.3"
-            // }
-            this._traceResolveDependency(packageJson.name, pnpmDotFolderPath, deployState);
-          } catch (resolveErr) {
-            if (resolveErr.code === 'MODULE_NOT_FOUND') {
-              // The workaround link isn't guaranteed to exist, so ignore if it's missing
-              // NOTE: If you encounter this warning a lot, please report it to the Rush maintainers.
-              console.log('Ignoring missing PNPM workaround link for ' + packageJsonFolderPath);
-            }
-          }
+  private _applyDependencyFilters(
+    allDependencyNames: Set<string>,
+    additionalDependenciesToInclude: string[] = [],
+    dependenciesToExclude: string[] = []
+  ): Set<string> {
+    console.log();
+    console.log('Additional dependencies to include:', additionalDependenciesToInclude);
+    console.log('Dependencies to exclude:', dependenciesToExclude);
+
+    dependenciesToExclude.forEach((patternWithStar) => {
+      allDependencyNames.forEach((dependency) => {
+        if (matchesWithStar(patternWithStar, dependency)) {
+          allDependencyNames.delete(dependency);
         }
-      }
-    }
+      });
+    });
+
+    additionalDependenciesToInclude.forEach((dependencyToInclude) => {
+      allDependencyNames.add(dependencyToInclude);
+    });
```

Comment:
I'm not sure if the logic needs to be more complicated than this. I assume we are going to leave it up to the individual to ensure that whatever additional packages they want are actually installed?

Reply:
I tested this case, and the error message is pretty understandable, especially since you are reporting the extra included packages.